network/server/httpjson/handlers.go

```diff
@@ -62,6 +62,7 @@ func RegisterHandlers(mux *http.ServeMux, service interface{}, opts ServerOption
 		// - methodName(RequestObject) error
 		// - methodName(RequestObject) (ResultObject, error)
 
+		// TODO(xichen): make HTTP server understand truncate method.
```

Comment:
If you add a `TruncateResult` as the return type for the RPC it'll get automatically registered.  Perhaps do that?